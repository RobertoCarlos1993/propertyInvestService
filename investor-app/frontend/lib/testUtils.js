const fakeProperties = [
  {
    id: "123",
    url: "https://www.idealista.com/inmueble/87821179/",
    price: 125000,
    size: 70,
    address: "Centro, Leganés"
  },
  {
    id: "124",
    url: "https://www.idealista.com/inmueble/89052114/",
    price: 299700,
    size: 152,
    address: "Casco Antiguo, Leganés"
  }
];

const typedValue = {
  price: 500,
  size: 80
};

export { fakeProperties, typedValue };
