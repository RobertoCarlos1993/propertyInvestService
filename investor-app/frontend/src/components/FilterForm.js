import React, { useState } from "react";
import styled from "styled-components";
const fakeProperties = [
  {
    id: "123",
    url: "https://www.idealista.com/inmueble/87821179/",
    price: 125000,
    size: 70,
    address: "Centro, Leganés"
  },
  {
    id: "124",
    url: "https://www.idealista.com/inmueble/89052114/",
    price: 299700,
    size: 152,
    address: "Casco Antiguo, Leganés"
  }
];

const Button = styled.button`
  background: ${props => (props.primary ? "grey" : "white")};
  color: ${props => (props.primary ? "white" : "palevioletred")};
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid grey;
  border-radius: 3px;
`;

const Input = styled.input`
  padding: 0.5em;
  margin: 0.5em;
  color: ${props => props.inputColor || "blue"};
  background: papayawhip;
  border: none;
  border-radius: 3px;
`;

const Select = styled.select`
  height: 35px;
  background: papayawhip;
  padding-left: 5px;
  font-size: 14px;
  border: none;
  margin-left: 10px;
`;

const Table = styled.table`
  border-spacing: 0;
  width: 100%;
  border: 1px solid ${props => props.theme.offWhite};
  thead {
    font-size: 10px;
  }
  td,
  th {
    border-bottom: 1px solid ${props => props.theme.offWhite};
    border-right: 1px solid ${props => props.theme.offWhite};
    padding: 5px;
    position: relative;
    &:last-child {
      border-right: none;
      width: 150px;
      button {
        width: 100%;
      }
    }
    label {
      padding: 10px 5px;
      display: block;
    }
  }
  tr {
    &:hover {
      background: ${props => props.theme.offWhite};
    }
  }
`;

const FilterForm = () => {
  const [query, setQuery] = useState({
    operation: "sale",
    price: "",
    propertyType: "flat",
    size: "",
    endType: "find"
  });

  const [properties, setProperties] = useState([]);

  const saveQueryParams = e => {
    console.log(e.target.value);
    setQuery({ ...query, [e.target.name]: e.target.value });
  };

  const callDB = () => {
    console.log("hi");
    setProperties(fakeProperties);
    // call endpoint
    /* const requestOptions = {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json"
      },
      redirect: "follow",
      body: JSON.stringify({ ...query })
    };
    fetch("/filterProperty", requestOptions)
      .then(response => response.text())
      .then(result => console.log(result))
      .catch(error => console.log("error", error));*/
  };

  return (
    <>
      <label>Operation:</label>
      <Select type="text" name="operation" onChange={saveQueryParams}>
        <option value="sale">Sale</option>
        <option value="rent">Rent</option>
      </Select>
      <label>Property Type:</label>
      <Select type="text" name="propertyType" onChange={saveQueryParams}>
        <option value="flat">Flat</option>
        <option value="chalet">Chalet</option>
        <option value="countryHouse">Country House</option>
        <option value="garage">Garage</option>
        <option value="office">Office</option>
      </Select>
      <label>Size:</label>
      <Input type="text" name="size" onChange={saveQueryParams} />
      <label>Price:</label>
      <Input
        type="text"
        name="price"
        onChange={saveQueryParams}
        value={query.price}
      />
      <label>Call Endpoint Type:</label>
      <Select type="text" name="endType" onChange={saveQueryParams}>
        <option value="findAll">Find All</option>
        <option value="find">Filter</option>
      </Select>
      <Button primary onClick={callDB} data-test="btnSubmit">
        Call DB
      </Button>
      {properties.length > 0 && (
        <Table data-test="propertiesTable">
          <thead>
            <tr>
              <th>Address</th>
              <th>Price</th>
              <th>Size</th>
              <th>Url</th>
            </tr>
          </thead>
          <tbody>
            {properties.map(property => (
              <tr key={property.id}>
                <td>{property.address}</td>
                <td>{property.price}</td>
                <td>{property.size}</td>
                <td>{property.url}</td>
              </tr>
            ))}
          </tbody>
        </Table>
      )}
    </>
  );
};

export default FilterForm;
