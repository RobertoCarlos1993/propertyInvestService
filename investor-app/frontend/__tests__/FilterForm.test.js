import React from "react";

import FilterForm from "../src/components/FilterForm";
import { fakeProperties, typedValue } from "../lib/testUtils";
import toJSON from "enzyme-to-json";
import { shallow, mount } from "../setupTest";

function typing(wrapper, name, value) {
  wrapper.find(`input[name="${name}"]`).simulate("change", {
    target: { name, value }
  });
}

describe("<FilterForm/>", () => {
  it("renders properly", () => {
    const wrapper = mount(<FilterForm />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("accepts typing a value and update state properly", () => {
    const wrapper = mount(<FilterForm />);
    typing(wrapper, "price", typedValue.price);
    wrapper.update();
    expect(wrapper.find(`input[name="price"]`).prop("value")).toEqual(
      typedValue.price
    );
  });

  it("Click on btn and spapshot!", () => {
    const wrapper = mount(<FilterForm />);
    const btn = wrapper.find('button[data-test="btnSubmit"]');
    btn.simulate("click");
    wrapper.update();
    const table = wrapper.find("table[data-test='propertiesTable']");
    expect(toJSON(table)).toMatchSnapshot();
  });

  it("Click on btn and verify table values", () => {
    const wrapper = mount(<FilterForm />);
    const btn = wrapper.find('button[data-test="btnSubmit"]');
    btn.simulate("click");
    wrapper.update();
    const table = wrapper.find("table[data-test='propertiesTable']");
    table.find("tr td:first-child").forEach((td, idx) => {
      expect(td.text()).toEqual("2");
    });
  });
});
