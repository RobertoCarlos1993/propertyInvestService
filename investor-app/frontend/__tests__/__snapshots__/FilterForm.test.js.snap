// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterForm/> Click on btn and spapshot! 1`] = `
<table
  className="sc-AxhCb kerOgc"
  data-test="propertiesTable"
>
  <thead>
    <tr>
      <th>
        Address
      </th>
      <th>
        Price
      </th>
      <th>
        Size
      </th>
      <th>
        Url
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="123"
    >
      <td>
        Centro, Leganés
      </td>
      <td>
        125000
      </td>
      <td>
        70
      </td>
      <td>
        https://www.idealista.com/inmueble/87821179/
      </td>
    </tr>
    <tr
      key="124"
    >
      <td>
        Casco Antiguo, Leganés
      </td>
      <td>
        299700
      </td>
      <td>
        152
      </td>
      <td>
        https://www.idealista.com/inmueble/89052114/
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<FilterForm/> renders properly 1`] = `
<FilterForm>
  <label>
    Operation:
  </label>
  <styled.select
    name="operation"
    onChange={[Function]}
    type="text"
  >
    <select
      className="sc-AxiKw dZlwWe"
      name="operation"
      onChange={[Function]}
      type="text"
    >
      <option
        value="sale"
      >
        Sale
      </option>
      <option
        value="rent"
      >
        Rent
      </option>
    </select>
  </styled.select>
  <label>
    Property Type:
  </label>
  <styled.select
    name="propertyType"
    onChange={[Function]}
    type="text"
  >
    <select
      className="sc-AxiKw dZlwWe"
      name="propertyType"
      onChange={[Function]}
      type="text"
    >
      <option
        value="flat"
      >
        Flat
      </option>
      <option
        value="chalet"
      >
        Chalet
      </option>
      <option
        value="countryHouse"
      >
        Country House
      </option>
      <option
        value="garage"
      >
        Garage
      </option>
      <option
        value="office"
      >
        Office
      </option>
    </select>
  </styled.select>
  <label>
    Size:
  </label>
  <styled.input
    name="size"
    onChange={[Function]}
    type="text"
  >
    <input
      className="sc-AxirZ opgZf"
      name="size"
      onChange={[Function]}
      type="text"
    />
  </styled.input>
  <label>
    Price:
  </label>
  <styled.input
    name="price"
    onChange={[Function]}
    type="text"
    value=""
  >
    <input
      className="sc-AxirZ opgZf"
      name="price"
      onChange={[Function]}
      type="text"
      value=""
    />
  </styled.input>
  <label>
    Call Endpoint Type:
  </label>
  <styled.select
    name="endType"
    onChange={[Function]}
    type="text"
  >
    <select
      className="sc-AxiKw dZlwWe"
      name="endType"
      onChange={[Function]}
      type="text"
    >
      <option
        value="findAll"
      >
        Find All
      </option>
      <option
        value="find"
      >
        Filter
      </option>
    </select>
  </styled.select>
  <styled.button
    data-test="btnSubmit"
    onClick={[Function]}
    primary={true}
  >
    <button
      className="sc-AxjAm haGGsl"
      data-test="btnSubmit"
      onClick={[Function]}
    >
      Call DB
    </button>
  </styled.button>
</FilterForm>
`;
